modules/javafx.graphics/src/main/java/com/sun/javafx/css/BitSet.java

```diff
@@ -533,16 +572,18 @@ public boolean equals(Object obj) {
         return true;
     }
 
-    abstract protected T getT(int index);
-    abstract protected int getIndex(T t);
+    protected abstract T getT(int index);
+    protected abstract int getIndex(T t);
 
-    /*
-     * Try to cast the arg to a T.
-     * @throws ClassCastException if the class of the argument is
-     *         is not a T
-     * @throws NullPointerException if the argument is null
+    /**
+     * Return {@code obj} if it is an instance of type {@code T},
+     * otherwise return {@code null}.
+     *
+     * @param obj the object to cast, cannot be {@code null}
+     * @return a type T, or {@code null} if the argument was not of this type
+     * @throws NullPointerException when {@code obj} is {@code null}
```

Comment:
Previously, this method always returned an instance of `T`. Now that is not the case, it might also simply return `null` if the argument passed into it is an instance of a different class. I think it makes sense to also return `null` when the argument passed into the method is `null`.

Reply:
I'm a bit unsure why that would be an improvement. Passing `null` to a function that doesn't expect it should IMHO never just return `null` but should instead be considered a programming error and result in a stack trace.  Passing in a non-null value that can't be casted is explicitly documented now that it would result in `null`.  One is a caller error, the other isn't IMHO (as the caller can't check if it is castable without another method -- I considered adding an `instanceof` method).

Or maybe I'm reading too much in to this and you are just pointing out that the function has changed from its previous contract -- I think this is okay as `BitSet` is not public API, nor are any of its subclasses.

Reply:
I found it strange that a method named `cast` would reject `null` as an argument, but will happily return `null` and require the caller to check it.

Reply:
The name might need updating, although when the name can't cover the exact intended functionality, the docs are supposed to explain all the details.

I may not need the cast at all anymore; I think the class is just asking for casts to satisfy some generic code, but that shouldn't be relevant.  All I would really need is an instanceof check.  I'll have another look.

Reply:
see comment about equals() above, should we store the element type and check against it when needed?

Reply:
@mstr2 I've changed this cast method now to a `getElementType` method to be able to provide a correct `equals` implementation under all circumstances.